src/java/org/apache/cassandra/index/sai/disk/v1/SegmentMetadata.java

```diff
@@ -64,6 +64,11 @@ public class SegmentMetadata implements Comparable<SegmentMetadata>
      */
     public final long segmentRowIdOffset;
 
+    /**
+     * Whether this segment is the last one in the sstable.
+     */
+    public final boolean isLastSegmentInSSTable;
+
```

Comment:
Instead of adding this variable here, used to decide how to adjust the min/max rowIds, could we maybe adjust those offsets when loading the metadata?